internal/cluster/cluster.go

```diff
@@ -211,6 +216,64 @@ func (m *Manager) DeregisterCluster(ctx context.Context, c *common.Cluster) erro
 	return err
 }
 
+func (m *Manager) reportValidationFailedMetrics(ctx context.Context, c *common.Cluster, ocpVersion, emailDomain string) error {
+	log := logutil.FromContext(ctx, m.log)
+	var validationRes validationsStatus
+	if c.ValidationsInfo != "" {
+		if err := json.Unmarshal([]byte(c.ValidationsInfo), &validationRes); err != nil {
+			log.WithError(err).Errorf("Failed to unmarshal validations info from cluster %s", *c.ID)
+			return err
+		}
+	}
+	for _, vRes := range validationRes {
+		for _, v := range vRes {
+			if v.Status == ValidationFailure {
+				m.metricAPI.ClusterValidationFailed(ocpVersion, emailDomain, models.ClusterValidationID(v.ID))
+			}
+		}
+	}
+	return nil
+}
+
+func (m *Manager) reportValidationStatusChanged(ctx context.Context, c *common.Cluster,
+	newValidationRes, currentValidationRes validationsStatus) {
+	for vCategory, vRes := range currentValidationRes {
+		for i, v := range vRes {
+			if newValidationRes[vCategory][i].Status == ValidationFailure && v.Status == ValidationSuccess {
+				m.metricAPI.ClusterValidationChanged(c.OpenshiftVersion, c.EmailDomain, models.ClusterValidationID(v.ID))
+				eventMsg := fmt.Sprintf("Cluster validation '%s' that used to succeed is now failing", v.ID)
+				m.eventsHandler.AddEvent(ctx, *c.ID, nil, models.EventSeverityWarning, eventMsg, time.Now())
+			}
+			if newValidationRes[vCategory][i].Status == ValidationSuccess && v.Status == ValidationFailure {
+				eventMsg := fmt.Sprintf("Cluster validation '%s' is now fixed", v.ID)
+				m.eventsHandler.AddEvent(ctx, *c.ID, nil, models.EventSeverityInfo, eventMsg, time.Now())
```

Comment:
Consider
```suggestion
			if newValidationRes[vCategory][i].Status != v.Status {
				m.metricAPI.ClusterValidationChanged(c.OpenshiftVersion, c.EmailDomain, models.ClusterValidationID(v.ID))
				eventMsg := fmt.Sprintf("Cluster validation '%s' status changed, new status: %s", v.ID, newValidationRes[vCategory][i].Status)
				EventSeverity := models.EventSeverityInfo
				if newValidationRes[vCategory][i].Status == ValidationFailure {
					EventSeverity := models.EventSeverityWarning
				}
				m.eventsHandler.AddEvent(ctx, *c.ID, nil, EventSeverity, eventMsg, time.Now())
			}
```

Reply:
In the change you describe we will generate events/metrics in those cases as well:
* pending --> success
* pending --> failure.

I don't think we want that.

Reply:
Don't we want event/metrics in case:
 - success -->  pending 
 - failure --> pending .

Reply:
This is not a real use case.

Reply:
Why not?
That's expected whenever a user will:
1. Delete or change VIP
2. Toggle the userManagedNetwork 
3. Toggle dhcpVIPsAllocation

Reply:
The main idea of the visibility epic, is to report issues, meaning `success` --> `failed`.

Other events where added only for UX, so they don't see a installation success with a lots of warnings that were not resolved in the events list.

What you described isn't an issue but the expected behaviour, it doesn't need to be reported to prometheus, maybe we can add events in this case for better UX but I think we should have a new task for it and make the change to host validations as well.